streaming/index.js

```diff
@@ -33,21 +33,14 @@ const redisUrlToClient = async (defaultConfig, redisUrl) => {
   let client;
 
   if (!redisUrl) {
-    client = redis.createClient(config);
+    client = new Redis(config);
   } else if (redisUrl.startsWith('unix://')) {
-    client = redis.createClient(Object.assign(config, {
-      socket: {
-        path: redisUrl.slice(7),
-      },
-    }));
+    client = new Redis(redisUrl.slice(7));
```

Comment:
Does this change mean that the other redis settings that come via config/env variables can't be overridden?

Reply:
I actually might refactor this a bit - are these specific ones we need to come through? I think ioredis supports everything we'd need https://github.com/redis/ioredis/blob/v4/API.md#new-redisport-host-options We'd just need to tweak the assignment where needed.

Reply:
Honestly, I'm not sure! I think everyone I know uses network sockets for connecting to redis, instead of unix sockets.

Main thing to look out for is being able to pass the password separately from the connection string, and setting any connection timeouts, I think

Reply:
Slightly tweaked this function and moved most of the configuration to the `redisConfigFromEnv()` function - had to shuffle some things around to work with the new library's options. Should be all good now :)